src/main/java/com/raf/si/Banka2Backend/services/UserService.java

```diff
@@ -1,15 +1,55 @@
 package com.raf.si.Banka2Backend.services;
 
+import com.raf.si.Banka2Backend.models.User;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
 @Service
-public class UserService implements UserDetailsService {
+public class UserService implements UserDetailsService, UserServiceInteface {
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-        //TODO: implement;
+        Optional<User> myUser = this.findByEmail(username);
+        if(myUser.isEmpty()) {
+            throw new UsernameNotFoundException("User with email: " + username + " not found");
+        }
+
+        return new org.springframework.security.core.userdetails.User(myUser.get().getEmail(), myUser.get().getPassword(), new ArrayList<>());
+    }
+
+    public Optional<User> findByEmail(String email) {
```

Comment:
Metode iz ovog servisa nisu implementirane, sve vracaju null ili empty. UserService bi trebalo da ima @autowired instancu UserRepository-ija i onda da se u metodama ove klase zovu metode repository-ja. @DusanGligorijevic Mozes to da dodas?